pkg/store/labelpb/label.go

```diff
@@ -292,30 +292,29 @@ func (m *ZLabel) Compare(other ZLabel) int {
 //
 // In case of existing labels already present in given label set, it will be overwritten by external one.
 // NOTE: Labels and extend has to be sorted.
-func ExtendSortedLabels(lset, extend labels.Labels) labels.Labels {
-	ret := make(labels.Labels, 0, len(lset)+len(extend))
-
-	// Inject external labels in place.
-	for len(lset) > 0 && len(extend) > 0 {
-		d := strings.Compare(lset[0].Name, extend[0].Name)
-		if d == 0 {
-			// Duplicate, prefer external labels.
-			// NOTE(fabxc): Maybe move it to a prefixed version to still ensure uniqueness of series?
-			ret = append(ret, extend[0])
-			lset, extend = lset[1:], extend[1:]
-		} else if d < 0 {
-			ret = append(ret, lset[0])
-			lset = lset[1:]
-		} else if d > 0 {
-			ret = append(ret, extend[0])
-			extend = extend[1:]
+// Taken from: https://github.com/prometheus/prometheus/blob/f2e02c52db/storage/remote/queue_manager.go#L899-L923
+func ExtendSortedLabels(ls labels.Labels, externalLabels []labels.Label) labels.Labels {
```

Comment:
This benchmarks slightly worse than the previous method but not terribly so: 

```
$ go test . -run XXX -bench BenchmarkExtendLabels -v
goos: linux
goarch: amd64
pkg: github.com/thanos-io/thanos/pkg/store/labelpb
cpu: 11th Gen Intel(R) Core(TM) i5-1135G7 @ 2.40GHz
BenchmarkExtendLabels
BenchmarkExtendLabels-8   	 1282364	       935.7 ns/op
PASS
ok  	github.com/thanos-io/thanos/pkg/store/labelpb	2.063s
```
vs
```
$ go test . -run XXX -bench BenchmarkExtendLabels -v
goos: linux
goarch: amd64
pkg: github.com/thanos-io/thanos/pkg/store/labelpb
cpu: 11th Gen Intel(R) Core(TM) i5-1135G7 @ 2.40GHz
BenchmarkExtendLabels
BenchmarkExtendLabels-8   	 1853936	       682.6 ns/op
PASS
ok  	github.com/thanos-io/thanos/pkg/store/labelpb	1.934s
```